docs/pages/contributing/documentation/how-to-contribute.mdx

```diff
@@ -1,168 +1,306 @@
-```code
-# If you are an external contributor, create a fork and push to it.
-# This command assumes that you have created an SSH key pair and stored
-# your public key in GitHub.
-$ git remote add fork ssh://git@github.com/my-user/teleport
-$ git rebase origin/master
-$ git push -u fork my-branch
-```
+1. Switch to the master branch by running the following command:
+    
+    ```code
+    $ git switch master
+    ```
 
-Visit https://github.com/gravitational/teleport and create a pull request (PR)
-against your branch. Our continuous integration (CI) pipeline will automatically
-assign reviewers to your PR. The best way to ensure that your PR is merged is to
-respond to reviewer feedback in a timely manner. If feedback goes unaddressed
-your PR may be closed due to inactivity. Please merge your PR as soon as
-possible once you have received all required approvals.
+    You can use a versioned branch if you want to update documentation for a specific previous version of Teleport.
+    However, it most cases, you should use the `master` branch.
 
-## Step 4/4 Backport your changes
+1. Create a new branch of `gravitational/teleport` for your changes by running a command similar to the following:
 
-Since our docs site is versioned, we want to ensure that any changes you make to
-the most recent version of our documentation are reflected for the versions of
+    ```code
+    $ git switch -c my-branch
+    ```
+
+    You now have a branch for a specific version of the documentation content.
+    Before making any changes, note how the content is organized so you know where to find the files to update.
+    
+    Within the `content/*.*` directory in your branch, there's a `docs` subdirectory. 
+    The `content/*.*/docs` subdirectory contains the files and folders used to produce a specific version of the documentation. 
+    The subdirectory includes the following:
+
+    * The `img/` folder for images used inside the pages.
+    * The `pages/` folder for the Markdown files that contain documentation content. 
+      Every file in this folder is rendered as a page.
+    * The `config.json` file for configuring version information.
+
+    You're now ready to make changes in your own branch using the code editor of your choice.
+
+## Step 3/5. Test changes locally
+
+As you make changes to the content in your local branch, it's helpful to see how the changes will be rendered when the documentation is published.
+
+You can use `yarn` to run a development server that automatically detects changes to the documentation as you make them and refreshes the page within the browser.
+You can also use `yarn` commands to check for errors and fix common issues.
+
+To test changes in your local environment:
+
+1. Change to the root of the `docs` directory by running the following command:
+    
+    ```code
+    $ cd docs
+    ```
+
+1. Start the development server by running the following commands:
+    
+    ```code
+    $ yarn install
+    $ yarn dev
+    ```
```

Comment:
Should the user have already run `cd docs` and `yarn install`?

Reply:
I didn't find it intuitive that I needed to run yarn/yarn install/yarn dev in the docs root directory rather than the docs (sub)directory I was working in. That's why I added the steps to reinforce context.